vms/event/views.py

```diff
@@ -1,78 +1,88 @@
 from django.core.urlresolvers import reverse
 from django.http import HttpResponse, HttpResponseRedirect
 from django.shortcuts import render
 from django.contrib import messages
-
+from braces.views import LoginRequiredMixin, AnonymousRequiredMixin
 from event.forms import EventForm, EventDateForm
+from django.views.generic.edit import FormView, UpdateView
+from django.views.generic.edit import DeleteView
+from django.views.generic import ListView
 from event.services import *
+from event.models import *
+from django.core.urlresolvers import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.shortcuts import render_to_response
+from django.http import Http404
+
+
+class AdministratorLoginRequiredMixin(object):
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        user = request.user
+        admin = None
+        try:
+            admin = user.administrator
+        except ObjectDoesNotExist:
+            pass
+        if not admin:
+            return render(request, 'vms/no_admin_rights.html')
+        else:
+            return super(AdministratorLoginRequiredMixin, self).dispatch(request, *args, **kwargs)
 
 
-@login_required
-def is_admin(request):
-    user = request.user
-    admin = None
-
-    try:
-        admin = user.administrator
-    except ObjectDoesNotExist:
-        pass
-
-    # check that an admin is logged in
-    if admin is not None:
-        return True
-    else:
-        return False
-
+class EventCreateView(LoginRequiredMixin, AdministratorLoginRequiredMixin, FormView):
+    template_name = 'event/create.html'
+    form_class = EventForm
 
-@login_required
-def create(request):
-    
-    if is_admin(request):
-        if request.method == 'POST':
-            form = EventForm(request.POST)
-            if form.is_valid():
-                start_date = form.cleaned_data['start_date']
-                if start_date < (datetime.date.today() - datetime.timedelta(days=1)):
-                    messages.add_message(request, messages.INFO, 'Start date should be today\'s date or later.')
-                    return render(request, 'event/create.html', {'form': form, })
-                else:
-                    form.save()
-                    return HttpResponseRedirect(reverse('event:list'))
-            else:
-                return render(request, 'event/create.html', {'form': form, })
+    def form_valid(self, form):
+        start_date = form.cleaned_data['start_date']
+        if start_date < (datetime.date.today() - datetime.timedelta(days=1)):
+            messages.add_message(self.request, messages.INFO, 'Start date should be today\'s date or later.')
+            return render(self.request, 'event/create.html', {'form': form,})
         else:
-            form = EventForm()
-            return render(request, 'event/create.html', {'form': form, })
-    else:
-        return render(request, 'vms/no_admin_rights.html')
```

Comment:
I have tested all these case manually. @smarshy Can we have few test cases for such cases, where the non admin should be redirected to no_admin_rights page?